### PR TITLE
itest: ensure LND gets correct CLI options when itest are parallel

### PR DESCRIPTION
**Pull request by aakselrod**
## Change Description

This PR ensures that when running itests in CI with `nativesql=true`, that option is passed correctly to LND. [Currently](https://github.com/lightningnetwork/lnd/actions/runs/10499052273/job/29085120784#step:6:253), `itest_parallel.sh` doesn't pass the option on to the per-tranche workers.

Note that this will cause the nativesql itests to fail, as there are test failures that have been masked by the fact that the tests aren't actually running with `nativesql=true`. #9022 fixes one, and several others will need to be fixed as well.

## Steps to Test
See the difference between the itests running on this PR vs. [on `master`](https://github.com/lightningnetwork/lnd/actions/runs/10499052273/job/29085120784#step:6:253).

### Testing
- [x] Your PR passes all CI checks.
- [x] Tests covering the positive and negative (error paths) are included.
- [x] Bug fixes contain tests triggering the bug to prevent regressions.

### Code Style and Documentation
- [x] The change obeys the [Code Documentation and Commenting](https://github.com/lightningnetwork/lnd/blob/master/docs/code_contribution_guidelines.md#CodeDocumentation) guidelines, and lines wrap at 80.
- [x] Commits follow the [Ideal Git Commit Structure](https://github.com/lightningnetwork/lnd/blob/master/docs/code_contribution_guidelines.md#IdealGitCommitStructure).
- [x] Any new logging statements use an appropriate subsystem and logging level.
- [x] Any new lncli commands have appropriate tags in the comments for the rpc in the proto file.
- [x] [There is a change description in the release notes](https://github.com/lightningnetwork/lnd/tree/master/docs/release-notes), or `[skip ci]` in the commit message for small changes.

📝 Please see our [Contribution Guidelines](https://github.com/lightningnetwork/lnd/blob/master/docs/code_contribution_guidelines.md) for further guidance.

### CODE REVIEW

File: scripts/itest_parallel.sh
Comment (guggero) [inline scripts/itest_parallel.sh:10]: nit: could be `shift 4`.

File: scripts/itest_parallel.sh
Comment (guggero) [inline scripts/itest_parallel.sh:21]: I think we should instead put `$(ITEST_FLAGS)` and `$(TEST_FLAGS)` in quotes in the makefile, as that's the way this was probably intended to work.

Comment (guggero) [reply scripts/itest_parallel.sh:21]: Or merge `TEST_FLAGS=$3` and `ITEST_FLAGS=$4` into just one. Because `$(TEST_FLAGS)` in the makefile might also contain multiple entries.

**Review by guggero — COMMENTED** [review]
Great catch! Thanks a lot for the fix. Although I think this requires a small patch to work fully with all scenarios.

**Comment by aakselrod** [comment]
updated this one by mistake, will fix

**Comment by Roasbeef** [comment]
Looks like this needs a `go mod tidy` to pass CI.

File: scripts/itest_parallel.sh
Comment (aakselrod) [reply scripts/itest_parallel.sh]: Changed to `shift 2` since I eliminated `TEST_FLAGS` and `ITEST_FLAGS`

File: scripts/itest_parallel.sh
Comment (aakselrod) [reply scripts/itest_parallel.sh]: I merged them the same way [itest_part.sh](https://github.com/lightningnetwork/lnd/blob/cc9e2b783ec983fbe3995d47d31898aa122b11c8/scripts/itest_part.sh#L6-L21) does it. Does this look better?

**Comment by aakselrod** [comment]
> Looks like this needs a `go mod tidy` to pass CI.

This was my bad. Lint passes now.

File: scripts/itest_parallel.sh
Comment (guggero) [inline scripts/itest_parallel.sh:7]: nit: text still says "4".

**Review by guggero — APPROVED** [review]
Thanks, LGTM :tada:

**Comment by guggero** [comment]
Needs a rebase because of the release notes.

**Review by yyforyongyu — APPROVED** [review]
Pending a rebase, otherwise LGTM, thanks for the fix!

**Comment by aakselrod** [comment]
Sorry I was out sick yesterday, will try to get a rebase in today

**Comment by aakselrod** [comment]
Rebase complete. I'll rebase #9022 after this one is merged.

**Pull request merged**
